extends KinematicBody2D

onready var SM = $StateMachine
onready var VP = get_viewport_rect()

#onready var global = get_node("/root/Global")
#onready var HUD = get_node("/root/Game/CanvasLayer/HUD")

onready var HUD = get_node("/root/Game/UI/HUD")
onready var Attack = load("res://Attack/Attack.tscn")



var velocity = Vector2.ZERO
var jump_power = Vector2.ZERO
var direction = 1

export var gravity = Vector2(0,30)

export var move_speed = 10
export var max_move = 150

export var jump_speed = 75
export var max_jump = 580

export var leap_speed = 75
export var max_leap = 575

var moving = false
var is_jumping = false


func _ready():
	pass


func _physics_process(_delta):
	velocity.x = clamp(velocity.x,-max_move,max_move)
	
	if direction < 0 and not $AnimatedSprite.flip_h: $AnimatedSprite.flip_h = true
	if direction > 0 and $AnimatedSprite.flip_h: $AnimatedSprite.flip_h = false
	
	if Input.is_action_just_pressed("attack"):
		var attack = Attack.instance()
		attack.position = position
		attack.position.x += 10 * direction
		attack.direction = direction
		get_node("/root/Game/Attack_Container").add_child(attack)
		

func is_moving():
	if Input.is_action_pressed("left") or Input.is_action_pressed("right"):
		return true
	return false

func move_vector():
	return Vector2(Input.get_action_strength("right") - Input.get_action_strength("left"),1.0)

func _unhandled_input(event):
	if event.is_action_pressed("left"):
		direction = -1
	if event.is_action_pressed("right"):
		direction = 1

func set_animation(anim):
	if $AnimatedSprite.animation == anim: return
	if $AnimatedSprite.frames.has_animation(anim): $AnimatedSprite.play(anim)
	else: $AnimatedSprite.play()

func die():
	HUD.update_lives(-1)
	queue_free()
	



